recipes/openldap/all/conanfile.py

```diff
@@ -1,114 +1,90 @@
+        self.requires("openssl/1.1.1m")
+        if self.options.with_cyrus_sasl:
             self.requires("cyrus-sasl/2.1.27")
 
-    def source(self):
-        conan_data = self.conan_data["sources"][self.version]
-        tools.get(url=conan_data["url"], sha256=conan_data["sha256"])
-        extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration(f"{self.name} is only supported on Linux")
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        yes_no = lambda v: "yes" if v else "no"
+        self._autotools = AutoToolsBuildEnvironment(self)
+        configure_args = [
+            "--enable-shared={}".format(yes_no(self.options.shared)),
+            "--enable-static={}".format(yes_no(not self.options.shared)),
+            "--with-cyrus_sasl={}".format(yes_no(self.options.with_cyrus_sasl)),
+            "--with-pic={}".format(yes_no(self.options.get_safe("fPIC", True))),
+            "--without-fetch",
+            "--with-tls=openssl",
+            "--enable-auditlog"]
+        self._configure_vars  = self._autotools.vars
+        self._configure_vars["systemdsystemunitdir"] = os.path.join(self.package_folder, "res")
+        self._autotools.configure(args=configure_args, configure_dir=self._source_subfolder, vars=self._configure_vars)
+        return self._autotools
 
     def build(self):
         for patch in self.conan_data["patches"][self.version]:
             tools.patch(**patch)
-        conf_args = []
-        autotools = AutoToolsBuildEnvironment(self)
-        if self.options.shared:
-            conf_args.extend(["--enable-shared", "--disable-static"])
-        else:
-            conf_args.extend(["--disable-shared", "--enable-static"])
-        if self.options.cyrus_sasl:
-            conf_args.append("--with-cyrus_sasl")
-        else:
-            conf_args.append("--without-cyrus_sasl")
-        conf_args.extend(
-            ["--without-fetch", "--with-tls=openssl", "--enable-auditlog"])
-
-        libpath = ""
-        for path in autotools.library_paths:
-            libpath += "{}:".format(path)
-        
-        # Need to link to -pthread instead of -lpthread for gcc 8 shared=True on CI job
-        libs = ""
-        autotools.libs.remove("pthread") 
-        for lib in autotools.libs:
-            libs += "-l{} ".format(lib)
-        libs += "-pthread "
-        with tools.environment_append({"systemdsystemunitdir": self.package_folder, "LD_LIBRARY_PATH": libpath, "LIBS": libs}):
-            autotools.configure(
-                configure_dir=os.path.join(
-                    self._source_subfolder),
-                args=conf_args)
-        self.run("make depend -j8")
-        autotools.make()
+        autotools = self._configure_autotools()
+        autotools.make(vars=self._configure_vars)
```

Comment:
self.run("make depend") is not needed ?

Reply:
autotools.make is enough. Avoid self.run, there are better helpers